backend/btrixcloud/crawls.py

```diff
@@ -612,6 +613,20 @@ async def update_crawl_state(self, crawl_id: str, state: str):
     async def shutdown_crawl(self, crawl_id: str, org: Organization, graceful: bool):
         """stop or cancel specified crawl"""
         result = None
+        redis_err = (
+            "Unable to connect to redis to set crawl-stop. Trying again in 1 second"
+        )
+
+        if graceful:
+            while True:
```

Comment:
Hm, probably shouldn't have a loop here, as the crawl could already be shutting down / redis might be down and may not ever restart.
Probably should handle this at the operator level which will get a regular callback.

Reply:
Actually, I think the change in operator is all that's needed!
The crawl_manager.shutdown_crawl()
will set stopping to true
https://github.com/webrecorder/browsertrix-cloud/blob/main/backend/btrixcloud/crawlmanager.py#L280

which will then be reflected in next sync:
https://github.com/webrecorder/browsertrix-cloud/blob/main/backend/btrixcloud/operator.py#L164

and so the `crawl.stopping` condition should be true:
https://github.com/webrecorder/browsertrix-cloud/blob/main/backend/btrixcloud/operator.py#L367

and that already handles retrying if redis fails, etc..

Reply:
Ah, nice! That keeps it good and simple :)